pkg/security/resolvers/tags/resolver.go

```diff
@@ -71,13 +72,15 @@ func (t *DefaultResolver) Start(ctx context.Context) error {
 
 // Resolve returns the tags for the given id
 func (t *DefaultResolver) Resolve(id string) []string {
-	tags, _ := t.tagger.Tag("container_id://"+id, types.OrchestratorCardinality)
+	entityID := types.NewEntityID(types.ContainerID, id)
+	tags, _ := t.tagger.Tag(entityID.String(), types.OrchestratorCardinality)
```

Comment:
why call `.String()` here ? Wouldn't it be more efficient if `Tag` was taking an EntityID directly ? It would allow to skip the string allocation altogether

Reply:
Indeed you are right. 

The blocker to do this is that the Tag method is exposed to python integrations via rtloader conversion.

Changing the signature of the Tag method implies a need to modify the rtloader conversion logic, and also all call to the Tag method in all python integrations (including custom user-implemented integrations). 

This change is not trivial, and the work done in the PR is already a lot, so we preferred to keep it this way for now, and we will work on optimising this in a next step. 

One solution that we were thinking about is to add a new method called `TagWithCompositeEntity` (we can think more about the name), which does the same as Tag, but receives an `EntityID` type, while we keep the `Tag` method in order not to break backward compatibility. And then we push all teams to use the new method and slowly deprecate the old one when it makes sense. 

Does it make sense for you?

Reply:
This makes total sense, thanks !